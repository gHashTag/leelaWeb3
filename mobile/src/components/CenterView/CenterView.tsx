import React, { ReactNode } from 'react'

import { StyleSheet, View, ViewStyle } from 'react-native'

const styles = StyleSheet.create({
  main: {
    alignItems: 'center',
    justifyContent: 'center',
  },
})

interface CenterViewProps {
  children: ReactNode
  flex?: number
  style?: ViewStyle
}

const CenterView = ({ children, flex = 1, style }: CenterViewProps) => {
  return (
    <View style={[styles.main, { flex }, style]} testID="center-view">
      {children}
    </View>
  )
}

export { CenterView }
