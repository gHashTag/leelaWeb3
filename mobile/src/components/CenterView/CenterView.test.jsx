import React from 'react'

import { Text } from 'react-native'

import { render } from '@testing-library/react-native'

import { CenterView } from './CenterView'

describe('CenterView Component', () => {
  it('renders children correctly', () => {
    const { toJSON } = render(
      <CenterView>
        <Text>Centered Content</Text>
      </CenterView>,
    )

    expect(toJSON()).toMatchSnapshot()
  })
  it('renders children orrectly', () => {
    const { getByText } = render(
      <CenterView>
        <Text>Centered Content</Text>
      </CenterView>,
    )

    const centeredContent = getByText('Centered Content')

    // Check if the children are rendered correctly
    expect(centeredContent).toBeTruthy()
  })
})
