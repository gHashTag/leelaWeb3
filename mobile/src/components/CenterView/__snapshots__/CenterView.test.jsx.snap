// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterView Component renders children correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {
        "flex": 1,
      },
      undefined,
    ]
  }
  testID="center-view"
>
  <Text>
    Centered Content
  </Text>
</View>
`;
